src/components/calculator.js

```diff
@@ -1,47 +1,67 @@
 import React from 'react';
 import '../App.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+
   }
 
+  onClickEvent = (event) => {
+    const { value } = event.target;
+    const { total, next, operation } = calculate(this.state, value);
+    this.setState({ total, next, operation });
+    const output = document.querySelector('.input');
+    if (next !== null) {
+      output.innerHTML = next;
+    } else if (total !== null) {
+      output.innerHTML = total;
+    } else {
+      output.innerHTML = '';
+    }
+  };
+
   render() {
     return (
       <div className="calculator">
         <div className="input-div">
-          <input className="input" />
+        <h2 className="input">0</h2>
         </div>
 
         <div className="row">
-          <button className="grey-btn" type="button">AC</button>
-          <button className="grey-btn" type="button">+/-</button>
-          <button className="grey-btn" type="button">%</button>
-          <button className="orange-btn" type="button">÷</button>
+          <button onClick={this.onClickEvent} className="grey-btn" type="button">AC</button>
+          <button onClick={this.onClickEvent} value="AC" className="grey-btn" type="button">+/-</button>
+          <button onClick={this.onClickEvent} value="+/-" className="grey-btn" type="button">%</button>
+          <button onClick={this.onClickEvent} value="÷" className="orange-btn" type="button">÷</button>
```

Comment:
- [ ] The values for your buttons do not match the text. For example when I click the `%` button I get the `+/-` operation because that is its `value`. Kindly refactor your button `value` attributes to match their text.

- [ ] Kindly add the `value` for the **`AC`** button.

- [ ] After the refactor, when `AC` is clicked, the display output is becomes empty, whereas it should display **`0`**. Kindly fix this. I think you set the `output.innerHTML` to `0`, instead of `''`.

Reply:
Thank you for the feedback. This has been resolved.